copy.go

```diff
@@ -21,19 +21,55 @@ import (
 	"fmt"
 
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	artifactspec "github.com/oras-project/artifacts-spec/specs-go/v1"
+	"golang.org/x/sync/semaphore"
 	"oras.land/oras-go/v2/content"
 	"oras.land/oras-go/v2/errdef"
 	"oras.land/oras-go/v2/internal/cas"
+	"oras.land/oras-go/v2/internal/docker"
 	"oras.land/oras-go/v2/internal/graph"
 	"oras.land/oras-go/v2/internal/status"
 )
 
+// defaultConcurrency is the default concurrency limit.
+// The value is consistent with dockerd and containerd.
+const defaultConcurrency = int64(3)
+
+// CopyOptions contains parameters for oras.Copy.
+type CopyOptions struct {
+	CopyGraphOptions
+	// ManifestFilter filters the manifest that is to be used as the root node
+	// for copy. Returns true if the manifest matches the filter, otherwise
+	// returns false.
+	ManifestFilter func(ctx context.Context, manifest ocispec.Descriptor) (bool, error)
```

Comment:
I'm re-considering this. We may still need two filter, one for the resolved manifest itself, another one for its down edges.